feat: changes

luzzif opened the pull request.

Commit:
feat(frontend): first version of creation form

Commit:
chore: upgrade dependencies

Commit:
fix(contracts): make the deployer contract the answerer account in the setup fork cct script

Commit:
fix(frontend): fix typo in specification type name

Commit:
fix(frontend): point to localhost:9080 to contact defillama answerer in dev mode

Commit:
feat(frontend): first draft oracle page implementation

guerrap commented on packages/contracts/.cct/setup-fork.js:
```diff
@@ -2,14 +2,11 @@ import { ChainId } from "@carrot-kpi/sdk";
 import { createRequire } from "module";
 import { fileURLToPath } from "url";
 import { getContract, parseUnits } from "viem";
+import { privateKeyToAddress } from "viem/accounts";
```
`ChainId` and `privateKeyToAddress` are not used.

luzzif replied:
Addressed in a27d86bea05fac63494443db2a86b72af5f16e6f

guerrap commented on packages/frontend/src/creation-form/payload-form/index.tsx:
```diff
@@ -0,0 +1,37 @@
+    if (!metric)
+        return (
+            <div className="flex justify-center">
+                <Typography>Select a metric</Typography>;
```
Left a `;` at the end of this line.

luzzif replied:
Addressed in a27d86bea05fac63494443db2a86b72af5f16e6f

guerrap commented on packages/frontend/src/creation-form/payload-form/tvl.tsx:
```diff
@@ -0,0 +1,105 @@
+                />
+            </div>
+            <div className="w-full md:w-1/2">
+                <DateTimeInput
```
Should we add a tooltip here? To explain what this timestamp is about?

luzzif replied:
Yes, that would be good. I've added it in a27d86bea05fac63494443db2a86b72af5f16e6f

guerrap commented on packages/frontend/src/creation-form/index.tsx:
```diff
@@ -1,12 +1,124 @@
+                payload,
+            };
+            const newState: State = { specification };
+
```
I think we should add a condition here, like the following:

```ts
if (
    !specification.metric ||
    !specification.payload?.protocol ||
    !specification.payload?.timestamp
)
    return;
```

With this, we'd avoid making HTTP calls to the `/validations` API, which we know would fail, since the payload is incomplete.

luzzif replied:
Nice catch, I've added the check in a27d86bea05fac63494443db2a86b72af5f16e6f

Commit:
chore: address pr comments

- remove unused imports
- avoid pointless calls to the specification validation service
- add tooltip to date time input for timestamp in tvl payload form
- remove extra semicolon
